python/Ganga/Core/GangaRepository/GangaRepositoryXML.py

```diff
@@ -1065,9 +1184,15 @@ def clean(self):
         rmrf(self.root)
         self.startup()
 
-    def updateIndexCache(self, obj):
+    def isObjectLoaded(self, obj):
         """
-        This method should be obsoleted fairly soon as it's no longer needed. We now handle caches correctly
+        This will return a true false if an object has been fully loaded into memory
+        Args:
+            obj (GangaObject): The object we want to know if it was loaded into memory
         """
-        pass
+        try:
+            _id = next(id_ for id_, o in self._fully_loaded.items() if o is obj)
+            return True
+        except StopIteration:
+            return False
```

Comment:
This is strange but I guess it's because the `list` `__contains__` method used equality rather than identity and out equality functions are slow and heavy.